tests/scripts/thread-cert/Cert_5_2_01_REEDAttach.py

```diff
@@ -0,0 +1,161 @@
+        msg.assertSentToDestinationAddress('ff02::1')
+        msg.assertMleMessageContainsTlv(mle.SourceAddress)
+        msg.assertMleMessageContainsTlv(mle.LeaderData)
+        msg.assertMleMessageContainsTlv(mle.Route64)
+
+        # 2 REED1: Attach to DUT_ROUTER1
+        self.nodes[REED1].start()
+        time.sleep(5)
+        self.assertEqual(self.nodes[REED1].get_state(), 'child')
+
+        # 3 DUT_ROUTER1: Verify MLE Rarent Response
+        router1_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER1)
+        msg = router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
+        msg.assertSentToNode(self.nodes[REED1])
+        msg.assertMleMessageContainsTlv(mle.Challenge)
+        msg.assertMleMessageContainsTlv(mle.Connectivity)
+        msg.assertMleMessageContainsTlv(mle.LeaderData)
+        msg.assertMleMessageContainsTlv(mle.LinkLayerFrameCounter)
+        msg.assertMleMessageContainsTlv(mle.LinkMargin)
+        msg.assertMleMessageContainsTlv(mle.Response)
+        msg.assertMleMessageContainsTlv(mle.SourceAddress)
+        msg.assertMleMessageContainsTlv(mle.Version)
+        msg.assertMleMessageContainsOptionalTlv(mle.MleFrameCounter)
+
+        # 4 DUT_ROUTER1: Verify MLE Child ID Response
+        msg = router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
+        msg.assertSentToNode(self.nodes[REED1])
+        msg.assertMleMessageContainsTlv(mle.Address16)
+        msg.assertMleMessageContainsTlv(mle.LeaderData)
+        msg.assertMleMessageContainsTlv(mle.NetworkData)
+        msg.assertMleMessageContainsTlv(mle.SourceAddress)
+        msg.assertMleMessageContainsOptionalTlv(mle.Route64)
+
+        # 5 Omitted
+
+        # 6 ED1: Attach to REED1
+        self.nodes[ED1].start()
+        time.sleep(20)
+        self.assertEqual(self.nodes[ED1].get_state(), 'child')
+
+        # 7 REED1: Verify Address Solicit Request to DUT_ROUTER1
+        reed1_messages = self.sniffer.get_messages_sent_by(REED1)
+        msg = reed1_messages.next_coap_message('0.02')
+        reed1_ipv6_address = msg.ipv6_packet.ipv6_header.source_address.compressed
+        msg.assertSentToNode(self.nodes[DUT_ROUTER1]);
+        msg.assertCoapMessageRequestUriPath('/a/as')
+
+        # 8 DUT_ROUTER1: Verify Address Solicit Requests to LEADER
+        router1_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER1)
+        msg = router1_messages.next_coap_message('0.02')
+        msg.assertSentToNode(self.nodes[LEADER]);
+        msg.assertCoapMessageRequestUriPath('/a/as')
+
+        # DUT_ROUTER1: Verify Address Solicit Response to REED1
+        msg = router1_messages.next_coap_message('2.04')
+        msg.assertSentToDestinationAddress(reed1_ipv6_address)
+
+        # 9 LEADER: Verify connectivity by sending an ICMPv6 Echo Request to REED1 
+        for addr in self.nodes[REED1].get_addrs():
+            if addr[0:4] != 'fe80':
+                self.assertTrue(self.nodes[LEADER].ping(addr))
```

Comment:
We have a method(get_ip6_address) in node_cli.py. You can get any address by this method.